src/features/surveys/rpc/getSurveyStats.ts

```diff
@@ -52,6 +53,12 @@ async function handle(
     const curDate = new Date(sortedSubmissions[0].submitted.slice(0, 10));
     const lastDate = new Date();
 
+    unlinkedCount = submissions.filter((sub) => {
+      if (sub.respondent !== null) {
+        return sub.respondent.id === undefined;
+      }
```

Comment:
At first, I tried thing that I tried
` unlinkedCount = submissions.filter(
      (sub) => sub.respondent?.id === undefined
    ).length;`
    
 but then the result was included people who has null for respondent. So with survey/1 it should return only 1 but instead, it return 5. 
 
 I just tried `unlinkedCount = submissions.filter((sub) => sub.respondent?.id).length;`
 and this one returns 7
 
 But survey/1 has only one person who is missing id as I know
 So I wrote it like that. I also thought that it would work like that but no idea why. Let me know if I misunderstanding something

Reply:
Ah of course, I mean the other way around: `!sub.respondent?.id` (i.e. _NOT_).

Reply:
umm it still returns 5 
![image](https://user-images.githubusercontent.com/77925373/224151021-937ce1e3-ac1f-4d6e-8aa1-3eb0f1646a91.png)

Reply:
Sorry for not thinking this through properly. I just gave it some proper thought, and what we want is to find all the submissions where there _is_ a respondent (i.e. non-anonymous) but where that respondent has no ID, i.e.:

```ts
unlinkedCount = submissions.filter(
  (sub) => sub.respondent && !sub.respondent.id
).length;
```

I have confirmed this to work.

Reply:
I thought `respondent?.id` would filter those submissions which is respondent : null.
![image](https://user-images.githubusercontent.com/77925373/224273276-c9fa52e1-a057-4f17-adb1-70101b231d1c.png)
How come that filter includes `respondent:null` ?